codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -378,6 +387,79 @@ void writeWindowUpdate(ChannelHandlerContext ctx) throws Http2Exception {
     }
 
     /**
+     * The local flow control state for a single stream that is not in a state where flow controlled frames cannot
+     * be exchanged.
+     */
+    private static final LocalFlowControlState REDUCED_FLOW_STATE = new LocalFlowControlState() {
+        @Override
+        public int windowSize() {
+            return 0;
+        }
+
+        @Override
+        public int initialWindowSize() {
+            return 0;
+        }
+
+        @Override
+        public void window(int initialWindowSize) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void incrementInitialStreamWindow(int delta) {
+            // This operation needs to be supported during the initial settings exchange when
+            // the peer has not yet acknowledged this peer being activated.
+        }
+
+        @Override
+        public void writeWindowUpdateIfNeeded(ChannelHandlerContext ctx) throws Http2Exception {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void consumeBytes(ChannelHandlerContext ctx, int numBytes) throws Http2Exception {
+            throw new UnsupportedOperationException();
```

Comment:
We may want to make this a no-op instead of unsupported (see previous comment).  Here's the scenario:

1. We receive data with `endOfStream=true`, which causes the stream to close.
2. We swap out flow state for `REDUCED_FLOW_STATE`
3. At some point later, the application finishes with the data and calls `consumeBytes`.
4. We throw because it's now unsupported.

WDYT?